File: swex/add-in/export-components/vb-net/ExportCommands.vb
Imports CodeStack.SwEx.AddIn.Attributes
Imports CodeStack.SwEx.AddIn.Enums
Imports CodeStack.SwEx.AddIn.Examples.SwExportComponent.My.Resources
Imports System.ComponentModel
Imports SolidWorks.Interop.swconst
Imports CodeStack.SwEx.Common.Attributes

<Title(GetType(Resources), NameOf(Resources.ToolbarTitle))>
<Description("Exports selected component to neutral format")>
<Icon(GetType(Resources), NameOf(Resources.export))>
Public Enum ExportCommands_e

	<Title("Export To Parasolid")>
	<Description("Exports selected component to parasolid")>
	<Icon(GetType(Resources), NameOf(Resources.export_parasolid))>
	<CommandItemInfo(True, True, swWorkspaceTypes_e.Assembly,
					 True, swCommandTabButtonTextDisplay_e.swCommandTabButton_TextBelow)>
	Parasolid

	<Title("Export To Iges")>
	<Description("Exports selected component to iges")>
	<Icon(GetType(Resources), NameOf(Resources.export_iges))>
	<CommandItemInfo(True, True, swWorkspaceTypes_e.Assembly,
					 True, swCommandTabButtonTextDisplay_e.swCommandTabButton_TextBelow)>
	Iges

	<Title("Export To Step")>
	<Description("Exports selected component to step")>
	<Icon(GetType(Resources), NameOf(Resources.export_step))>
	<CommandItemInfo(True, True, swWorkspaceTypes_e.Assembly,
					 True, swCommandTabButtonTextDisplay_e.swCommandTabButton_TextBelow)>
	[Step]

End Enum

File: swex/add-in/export-components/csharp/ExportCommands.cs
using CodeStack.SwEx.AddIn.Attributes;
using CodeStack.SwEx.AddIn.Enums;
using CodeStack.SwEx.AddIn.Examples.SwExportComponent.Properties;
using CodeStack.SwEx.Common.Attributes;
using SolidWorks.Interop.swconst;
using System.ComponentModel;

namespace CodeStack.SwEx.AddIn.Examples.SwExportComponent
{
    [Title(typeof(Resources), nameof(Resources.ToolbarTitle))]
    [Description("Exports selected component to neutral format")]
    [Icon(typeof(Resources), nameof(Resources.export))]
    public enum ExportCommands_e
    {
        [Title("Export To Parasolid")]
        [Description("Exports selected component to parasolid")]
        [Icon(typeof(Resources), nameof(Resources.export_parasolid))]
        [CommandItemInfo(true, true, swWorkspaceTypes_e.Assembly,
            true, swCommandTabButtonTextDisplay_e.swCommandTabButton_TextBelow)]
        Parasolid,

        [Title("Export To Iges")]
        [Description("Exports selected component to iges")]
        [Icon(typeof(Resources), nameof(Resources.export_iges))]
        [CommandItemInfo(true, true, swWorkspaceTypes_e.Assembly,
            true, swCommandTabButtonTextDisplay_e.swCommandTabButton_TextBelow)]
        Iges,

        [Title("Export To Step")]
        [Description("Exports selected component to step")]
        [Icon(typeof(Resources), nameof(Resources.export_step))]
        [CommandItemInfo(true, true, swWorkspaceTypes_e.Assembly,
            true, swCommandTabButtonTextDisplay_e.swCommandTabButton_TextBelow)]
        Step
    }
}
